### PR TITLE
Build onboarding network functions page

### PR DESCRIPTION
**Pull request by carkod**
## Done

Onboarding network functions page, part of /telco

## QA

- Check https://ubuntu-com-9884.demos.haus/telco/osm/onboarding-network-functions against [copy docs](https://docs.google.com/document/d/1_35nO3riTFlSFSp2B7FFdl60cjZe7Owwv0bcjY1s8Kg/edit#)

Design:

image

## Issue / Card

Fixes canonical-web-and-design/web-squad#4154

### CODE REVIEW

File: templates/telco/osm/onboarding-network-functions.html [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+</section>
+
+<section class="p-strip--suru-accent is-dark">
+  <div class="row">
+    
+    <div class="col-12">
+      <div class="p-heading-icon--small">
+        <h2>Tell us about your network function</h2>
+        <p>Canonical can help you automate your network function onboarding to Charmed OSM. Please choose the relevant options for your network functions to see how it can work for you.</p>
+        <p><a href="https://charmed-osm.com/#get-in-touch" class="p-button--positive">Start now</a></p>
```
Comment (anthonydillon) [inline templates/telco/osm/onboarding-network-functions.html:145]: This needs external icon

File: templates/telco/osm/onboarding-network-functions.html [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+		<h2>Let us help you in onboarding</h2>
+    <div class="col-6 p-card">
+      <h4><a class="p-link--external" href="https://charmed-osm.com/">Charmed OSM deployment support</a></h4>
+			<hr />
+      <ul class="p-list">
+        <li class="p-list__item is-ticked">Supported with <a href="/ua">Ubuntu Advantage</a></li>
+        <li class="p-list__item is-ticked">Availability of managed and professional services</li>
+        <li class="p-list__item is-ticked">ESM - Security updates for Ubuntu LTS for additional 3-5 years</li>
+      </ul>
+    </div>
```
Comment (anthonydillon) [inline templates/telco/osm/onboarding-network-functions.html:162]: Could you use the correct card classes please: https://vanillaframework.io/docs/patterns/card#header

File: templates/telco/osm/onboarding-network-functions.html [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+  </div>
+</section>
+
+<section class="p-strip--light">
+  <div class="row">
+    <div class="col-6">
+      <h2>Learn onboarding from scratch</h2>
+    </div>
+    <div class="col-6 u-embedded-media u-hide--small">
```
Comment (anthonydillon) [inline templates/telco/osm/onboarding-network-functions.html:179]: Why hide this on small screens?

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Quiet a few things here @carkod. Comments in code and a few QA issues:

## QA
- The hero image is not the right dimensions: 
image
- The "Charmed OSM deployment support" section has part link in the header in the copy doc. No link in the design and you have it as a full link. I would recommend removing the heading link and add a link to the OSM site below these cards.

File: templates/telco/osm/onboarding-network-functions.html (hunk 3 above)
Comment (carkod) [reply templates/telco/osm/onboarding-network-functions.html:179]: We seem to hide most images in small screens which I found it kind of odd. But yes, I guess this is a video shouldn't have this class

File: templates/telco/osm/onboarding-network-functions.html [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+    <h2>3 easy steps to onboard your network function</h2>
+		<p>Onboarding isall about configuring the day 0, day 1 and day 2 operations. All the life cycle stages can be achieved by defining these operations properly in the OSM packages for the network functions. The process includes:</p>
+		<hr style="margin-bottom: 2rem;" />
+		<div class="row">
+			<div class="col-6">
+				<h2>Day 0</h2>
+			</div>
+			<div class="col-6">
+				<ul>
+					<li>Create the network function descriptors as templates to instantiate/terminate the service.</li>
```
[the remark this answers is not in the document]
Comment (carkod) [reply templates/telco/osm/onboarding-network-functions.html:57]: That adds too much space and the design doesn't have any space between the `li` elements
spacing-network-functions

**Comment by carkod** [comment]
> Quiet a few things here @carkod. Comments in code and a few QA issues:
> 
> ## QA
> * The hero image is not the right dimensions:
>   image
> * The "Charmed OSM deployment support" section has part link in the header in the copy doc. No link in the design and you have it as a full link. I would recommend removing the heading link and add a link to the OSM site below these cards.

Addressed most of the comments. Still, there are missing graphs in the copy docs, changes requested to the graph and the entire section of hypervisors is missing.

File: templates/telco/osm/onboarding-network-functions.html (hunk 2 above)
Comment (carkod) [reply templates/telco/osm/onboarding-network-functions.html]: I added the `card-title` I believe that's what you meant? `p-list` seems to be fine

File: templates/telco/osm/onboarding-network-functions.html [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+    <div class="col-6">
+      <h1>Onboarding Network Functions</h1>
+      <p><a class="p-link--external" href="https://osm.etsi.org/" alt="">Open Source MANO</a> provides a platform to easily migrate your traditional telco functions to virtualized and cloud-native environments. It supports onboarding of all types of network functions:</p>
+      <ul>
+          <li>Virtualized Network Functions (VNF)</li>
+          <li>Cloud-native Network Functions (CNF)</li>
+          <li>Physical Network Functions (PNF)</li>
+      </ul>
+      <p>
+        <a class="p-button--positive" href="#get-in-touch">
```
Comment (anthonydillon) [inline templates/telco/osm/onboarding-network-functions.html:21]: This href should link to a contact-us page with the class `js-invoke-modal` to bring up the modal

File: templates/telco/osm/onboarding-network-functions.html [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+        loading="auto"
+        ) | safe
+      }}
+    </div>
+  </div>
+</section>
+
+<section class="p-strip--light is-bordered">
+  <div class="row">
+    <h2>3 easy steps to onboard your network function</h2>
```
Comment (anthonydillon) [inline templates/telco/osm/onboarding-network-functions.html:47]: This content is not within a column. I would suggest a `col-8`

File: templates/telco/osm/onboarding-network-functions.html [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+		<p>Onboarding is all about configuring the day 0, day 1 and day 2 operations. All the life cycle stages can be achieved by defining these operations properly in the OSM packages for the network functions. The <a href="https://osm.etsi.org/docs/vnf-onboarding-guidelines/" class="p-link--external">process</a> includes:</p>
+		<hr  />
+		<div class="row">
```
Comment (anthonydillon) [inline templates/telco/osm/onboarding-network-functions.html:50]: You have nested rows here?

File: templates/telco/osm/onboarding-network-functions.html [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+        width="1000",
+        height="180",
+        hi_def=True,
+        loading="auto"
+        ) | safe
+      }}
+    </div>
+</section>
+
+<section class="p-strip">
```
[the remark this answers is not in the document]
Comment (anthonydillon) [reply templates/telco/osm/onboarding-network-functions.html:104]: This was not actioned

File: navigation.yaml [hunk 9]
```diff
@@ -1621,6 +1621,16 @@ ubuntu1604:
...
 
+telco:
+  title: Telco
+  path: /telco
+
+  children:
+    - title: Overview
+      path: /telco
+    - title: Onboarding Network Functions
+      path: /telco/osm/onboarding-network-functions
```
Comment (anthonydillon) [inline navigation.yaml:1632]: This doesn't seem right. There is no `/telco/osm` page. @pmahnke

**Comment by anasereijo** [comment]
@carkod

can you also start putting the link to designs on the QA steps? I didn't realise we have a design for this and people reviewing it should be looking into that + copydoc. Thanks

See changes below:

- [ ] hero section:

Screenshot 2021-06-16 at 12 03 33

- [ ] "3 easy steps to onboard your network function" section"
Screenshot 2021-06-16 at 12 14 39

- [ ] "Lifecycle management" section:
apparently there are some copy changes to be done on this graph. I asked @SirSamTumless on the copydoc to get in touch with Brand
Screenshot 2021-06-16 at 12 18 22


- [ ] "VNF onboarding in OSM Hackfests" section

awaiting for a link with the image
Screenshot 2021-06-16 at 12 19 22

Screenshot 2021-06-16 at 12 21 41

- [ ] Let us help you in onboarding
Screenshot 2021-06-16 at 12 23 16

File: templates/telco/osm/onboarding-network-functions.html [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+  <div class="row u-vertically-center">
+    <div class="col-12">
+      <h2>Lifecycle management</h2>
+      <p>The goal of onboarding is to create a package which can fulfil the stages to manage the lifecycle of network services. The package should  include network function instantiation, scaling in/out, healing or recovery, updating, operating and termination.</p>
+    </div>
+		<hr  />
```
Comment (anthonydillon) [inline templates/telco/osm/onboarding-network-functions.html:110]: We have [the `p-separator`](https://vanillaframework.io/docs/base/separators#separator) for this.

File: templates/telco/osm/onboarding-network-functions.html [hunk 11]
```diff
@@ -0,0 +1,191 @@
...
+    <div class="col-6 p-card">
+      <h3 class="p-card__title">Network function onboarding support </h3>
+			<hr />
+      <ul class="p-list">
+        <li class="p-list__item is-ticked">Support for automation of the network function</li>
+        <li class="p-list__item is-ticked">Onboarding it to Charmed OSM</li>
+      </ul>
+    </div>
+    <p>
+      *&nbsp;<a class="p-link--external" href="https://charmed-osm.com/">https://charmed-osm.com/</a>
```
Comment (anthonydillon) [inline templates/telco/osm/onboarding-network-functions.html:173]: Where did this come from? This is not a foot note but a link for the user to learn more.

File: templates/telco/osm/onboarding-network-functions.html [hunk 12]
```diff
@@ -0,0 +1,191 @@
...
+  </div>
+</section>
+
+<section class="p-strip--suru-accent is-dark">
+  <div class="row">
+    <div class="col-12">
+      <div class="p-heading-icon--small">
+        <h2 id="get-in-touch">Tell us about your network function</h2>
+        <p>Canonical can help you automate your network function onboarding to Charmed OSM. Please choose the relevant options for your network functions to see how it can work for you.</p>
+        <p><a href="https://charmed-osm.com/#get-in-touch" class="p-button--positive p-external--link">Start now</a></p>
```
Comment (anthonydillon) [inline templates/telco/osm/onboarding-network-functions.html:146]:
```suggestion
        <p><a href="https://charmed-osm.com/#get-in-touch" class="p-button--positive p-link--external">Start now</a></p>
```

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Some more comments in the code

File: templates/telco/osm/onboarding-network-functions.html [hunk 13]
```diff
@@ -0,0 +1,191 @@
...
+{% block meta_description %}Onboarding network functions with Canonical Open Source MANO.{% endblock meta_description %}
+
+{% block meta_copydoc %}https://docs.google.com/document/d/1_35nO3riTFlSFSp2B7FFdl60cjZe7Owwv0bcjY1s8Kg/edit#{% endblock meta_copydoc %}
+
+{% block content %}
+<section class="p-strip--suru-topped is-deep is-bordered">
+  <div class="row u-equal-height">
+    <div class="col-6">
+      <h1>Onboarding Network Functions</h1>
+      <p><a class="p-link--external" href="https://osm.etsi.org/" alt="">Open Source MANO</a> provides a platform to easily migrate your traditional telco functions to virtualized and cloud-native environments. It supports onboarding of all types of network functions:</p>
```
Comment (anthonydillon) [inline templates/telco/osm/onboarding-network-functions.html:14]: `alt` is not allowed on an anchor

File: templates/telco/osm/onboarding-network-functions.html (hunk 5 above)
Comment (carkod) [reply templates/telco/osm/onboarding-network-functions.html:21]: In the copy doc this is linking to "Tell us about your network function" and it has been confirmed that's how they want it.

File: templates/telco/osm/onboarding-network-functions.html (hunk 10 above)
Comment (carkod) [reply templates/telco/osm/onboarding-network-functions.html:110]: `p-separator` is giving too much space compared to [the design](https://github.com/canonical-web-and-design/web-squad/issues/4153)?

File: templates/telco/osm/onboarding-network-functions.html (hunk 11 above)
Comment (carkod) [reply templates/telco/osm/onboarding-network-functions.html]: Link in the header, as mentioned by @anasereijo

File: templates/telco/osm/onboarding-network-functions.html (hunk 10 above)
Comment (carkod) [reply templates/telco/osm/onboarding-network-functions.html]: @anasereijo Using `p-separator` seems to do the job of adding more space. Are you ok with this?

File: templates/telco/osm/onboarding-network-functions.html (hunk 7 above)
Comment (carkod) [reply templates/telco/osm/onboarding-network-functions.html]: Fixed

**Comment by carkod** [comment]
> @carkod
> 
> can you also start putting the link to designs on the QA steps? I didn't realise we have a design for this and people reviewing it should be looking into that + copydoc. Thanks
> 
> See changes below:
> 
> * [x]  hero section:
> 
> Screenshot 2021-06-16 at 12 03 33
> 
> * [x]  "3 easy steps to onboard your network function" section"
> 
> Screenshot 2021-06-16 at 12 14 39
> 
> * [x]  "Lifecycle management" section:
>   apparently there are some copy changes to be done on this graph. I asked @SirSamTumless on the copydoc to get in touch with Brand
> 
> Screenshot 2021-06-16 at 12 18 22
> 
> * [x]  "VNF onboarding in OSM Hackfests" section
> 
> awaiting for a link with the image
> Screenshot 2021-06-16 at 12 19 22
> 
> Screenshot 2021-06-16 at 12 21 41
> 
> * [x]  Let us help you in onboarding
> 
> Screenshot 2021-06-16 at 12 23 16

I don't know where are the links to the design. i used the design attached to the [issue](https://github.com/canonical-web-and-design/web-squad/issues/4153). 

For spacing in  "3 easy steps to onboard your network function" section" and "Lifecycle management" section, I used what Ant suggested `p-separator` hopefully is fine? or too much space? If it's too much, then I would like to know the specific number of pixels we have in the design to do something custom.

Everything else I believe it's fixed.

**Comment by anasereijo** [comment]
@carkod we have now the image to use:

https://assets.ubuntu.com/v1/a1afcfd0-V.jpg

use the class `p-image--bordered`

**Comment by anasereijo** [comment]
@carkod when I say link to the design can be either a zeplin file or a link to the issue if it's just a jpg.

Thanks for making the changes. You missed one thing:

Screenshot 2021-06-16 at 14 11 39

**Comment by carkod** [comment]
> @carkod when I say link to the design can be either a zeplin file or a link to the issue if it's just a jpg.
> 
> Thanks for making the changes. You missed one thing:
> 
> Screenshot 2021-06-16 at 14 11 39

Done

**Comment by anasereijo** [comment]
lgtm

**Comment by SirSamTumless** [comment]
@anthonydillon Can you re-review this when you get a chance?

File: navigation.yaml (hunk 9 above)
Comment (anthonydillon) [reply navigation.yaml:1632]: What do you think @pmahnke?

File: templates/telco/osm/onboarding-network-functions.html [hunk 14]
```diff
@@ -0,0 +1,205 @@
...
+        width="1000",
+        height="500",
+        hi_def=True,
+        loading="lazy"
+        ) | safe
+      }}
+    </div>
+  </div>
+</section>
+<hr />
```
Comment (anthonydillon) [inline templates/telco/osm/onboarding-network-functions.html:129]: You can remove this as you have `is-bordered` above.

File: templates/telco/osm/onboarding-network-functions.html [hunk 15]
```diff
@@ -0,0 +1,205 @@
...
+      </p>
+    </div>
+  </div>
+</section>
+
+<section class="p-strip--suru-accent is-dark">
+  <div class="row">
+    <div class="col-12">
+      <div class="p-heading-icon--small">
+        <h4 id="get-in-touch">Tell us about your network function</h4>
```
Comment (anthonydillon) [inline templates/telco/osm/onboarding-network-functions.html:161]: This hash is used by the dynamic forms which will become confusing. Can you use a different hash please.

File: templates/telco/osm/onboarding-network-functions.html [hunk 16]
```diff
@@ -0,0 +1,205 @@
...
+        <p>Canonical can help you automate your network function onboarding to Charmed OSM. Please choose the relevant options for your network functions to see how it can work for you.</p>
+        <p><a href="https://charmed-osm.com/#get-in-touch" class="p-button--positive p-link--external">Start now</a></p>
+      </div>
+    </div>
+  </div>
+</section>
+
+<section class="p-strip">
+  <div class="row u-equal-height">
+		<h2>Let us help you in onboarding</h2>
```
Comment (anthonydillon) [inline templates/telco/osm/onboarding-network-functions.html:171]: This element is not within a column

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Some questions/comments around the use of `get-in-touch`

File: navigation.yaml (hunk 9 above)
Comment (carkod) [reply navigation.yaml:1632]: I believe the `/telco/osm` page will come soon, @SirSamTumless

File: templates/telco/osm/onboarding-network-functions.html (hunk 15 above)
Comment (carkod) [reply templates/telco/osm/onboarding-network-functions.html]: Updated to #tell-us

**Review by anthonydillon — APPROVED** [review]
Thanks @carkod :+1:

**Pull request merged**